cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java

```diff
@@ -604,13 +604,55 @@ public TeleDecl.FnDecl fnDecl(@NotNull GenericNode<?> node, Stmt.Accessibility a
           })
           .toImmutableSeq());
     }
-    // TODO: implement this
     if (node.is(ARRAY_EXPR)) {
-      return new Expr.HoleExpr(pos, false, null);
+      var arrayBlock = node.peekChild(ARRAY_BLOCK);
+
+      if (arrayBlock != null) {
+        if (arrayBlock.is(ARRAY_COMP_BLOCK)) {
```

Comment:
- extract two helper methods: `arrayCompBlock(GenericNode)` and `arrayElementList(GenericNode)`
- return early when `arrayBlock == null`

Reply:
Yes and braces can be omitted like
```
if (arrayBlock == null) return ...;
if (arrayBlock.is(...)) return arrayCompBlock(arrayBlock);
if (arrayBlock.is(...)) return arrayElementList(arrayBlock);

return unreachable(arrayBlock);
```